File: backend/ReservaTuCitaYa/ReservaTuCitaYa.Domain/Entities/Cliente.cs
using ReservaTuCitaYa.Domain.Common;
using ReservaTuCitaYa.Domain.Enums;

namespace ReservaTuCitaYa.Domain.Entities;

public sealed class Cliente : BaseEntity
{
    public Guid OrganizacionId { get; set; }
    public TipoDocumento TipoDocumento { get; set; }
    public string NumeroDocumento { get; set; } = string.Empty;
    public string Nombres { get; set; } = string.Empty;
    public string Apellidos { get; set; } = string.Empty;
    public string? Correo { get; set; }
    public string? Telefono { get; set; }
    public string? Direccion { get; set; }
    public DateOnly? FechaNacimiento { get; set; }
    public string? Observaciones { get; set; }

    public Organizacion Organizacion { get; set; } = null!;
}

File: frontend/ReservaTuCitaYa/src/features/clientes/types/Cliente.ts
export interface ClienteListado {
  id: number;
  nombres: string;
  apellidos: string;
  tipoDocumento: string;
  numeroDocumento: string;
  correo: string;
  telefono: string;
  estado: boolean;
}

export interface ClienteDetalle extends ClienteListado {
  direccion: string;
  fechaNacimiento: string;
  observaciones?: string;
  organizacionId: number;
}

export interface CrearClienteRequest {
  tipoDocumento: string;
  numeroDocumento: string;
  nombres: string;
  apellidos: string;
  correo: string;
  telefono: string;
  direccion: string;
  fechaNacimiento: string;
  observaciones?: string;
}

export interface ActualizarClienteRequest {
  id: number;
  tipoDocumento: string;
  numeroDocumento: string;
  nombres: string;
  apellidos: string;
  correo: string;
  telefono: string;
  direccion?: string;
  fechaNacimiento?: string;
  observaciones?: string;
  estado: boolean;
}


export interface ClienteFiltros {
  estado?: boolean;
  busqueda?: string;
  pagina?: number;
  tamañoPagina?: number;
}

export interface PaginaResultado<T> {
  total: number;
  paginaActual: number;
  tamañoPagina: number;
  elementos: T[];
}

